### ISSUE TITLE
tablets: Scylla fails on restart after tablet migration due to resharding attempt

### ISSUE THREAD

**Issue by tgrabiec** [scylladb/scylladb#14937]
Example error:
```
[shard 0] init - Startup failed: std::runtime_error (Exception while populating keyspace 'scylla_bench' with column family 'test' from file '/var/lib/scylla/data/scylla_bench/test-12e75630318b11ee9bca9d0867272343': seastar::internal::backtraced<std::runtime_error> (Attempted to obtain static sharder for table scylla_bench.test Backtrace: 0x5bc038e 0x5bc0950 0x5bc0c28 0x4f56395 0x4f560a4 0x57f467f 0x1e73039 0x3be5858 0x445cfbe 0x445e07f 0x12f61ea 0x5824aef 0x5825d27 0x5825099 0x57c85e7 0x57c779c 0x127c5ff 0x127e070 0x127ab8e /opt/scylladb/libreloc/libc.so.6+0x27b49 /opt/scylladb/libreloc/libc.so.6+0x27c0a 0x1278924
```

After tablet is no longer owned by a node, its token range may no longer be present on the same shard as before according to the sharder (shard 0 will be used). This makes the sstable appear as requiring resharding, which is not implemented for tablets yet.  

The issue will go away when integration with compaction groups is implemented, because sstables of migrated tablets will be recognized as belonging to a single shard and will not need resharding.

**Comment by avikivity** [comment]
Or we can just ignore unowned token ranges, and only reshard when an sstable has owned ranges on more than one shard.

**Comment by tgrabiec** [comment]
Fixed by bd8bb5d4b157af07949d644626d8c1622b34326d.
